operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/EventDispatcher.java

```diff
@@ -219,7 +219,7 @@ public R updateStatus(R resource) {
       return resourceOperation
           .inNamespace(resource.getMetadata().getNamespace())
           .withName(getName(resource))
-          .updateStatus(resource);
+          .patchStatus(resource);
```

Comment:
@metacosm I think we should discuss this more in depth under an issue. I think we need both, patch and update status without optimistic locking. Will create an issue and describe the use cases.

Reply:
Can we make this a separate PR from the metric and refactoring?